packages/block-library/src/post-author/edit.js

```diff
@@ -1,31 +1,256 @@
+/**
+ * External dependencies
+ */
+import { forEach } from 'lodash';
+import classnames from 'classnames';
+
 /**
  * WordPress dependencies
  */
-import { useEntityProp, useEntityId } from '@wordpress/core-data';
+import apiFetch from '@wordpress/api-fetch';
+import { useRef, useState } from '@wordpress/element';
+import { useEntityProp } from '@wordpress/core-data';
+import {
+	AlignmentToolbar,
+	BlockControls,
+	FontSizePicker,
+	InspectorControls,
+	RichText,
+	__experimentalUseColors,
+	BlockColorsStyleSelector,
+	withFontSizes,
+} from '@wordpress/block-editor';
+import { PanelBody, SelectControl, ToggleControl } from '@wordpress/components';
 import { useSelect } from '@wordpress/data';
-import { sprintf, __ } from '@wordpress/i18n';
-
-function PostAuthorDisplay() {
-	const [ authorId ] = useEntityProp( 'postType', 'post', 'author' );
-	const author = useSelect(
-		( select ) =>
-			select( 'core' ).getEntityRecord( 'root', 'user', authorId ),
-		[ authorId ]
+import { __ } from '@wordpress/i18n';
+
+const DEFAULT_AVATAR_SIZE = 24;
+
+async function getAuthorDetails( authorId, setAuthorDetails ) {
+	const authorDetails = await apiFetch( {
+		path: '/wp/v2/users/' + authorId + '?context=edit',
+	} );
+	setAuthorDetails( authorDetails );
+}
+
+function PostAuthorDisplay( {
+	postAuthor,
+	setPostAuthor,
+	authorDetails,
+	setAuthorDetails,
+	props,
+} ) {
+	const ref = useRef();
+
+	const { authors } = useSelect( ( select ) => {
+		const { getAuthors } = select( 'core' );
+		return {
+			authors: getAuthors(),
+		};
+	} );
+
+	const { isSelected, fontSize, setFontSize } = props;
+	const {
+		TextColor,
+		BackgroundColor,
+		InspectorControlsColorPanel,
+		ColorPanel,
+	} = __experimentalUseColors(
+		[
+			{ name: 'textColor', property: 'color' },
+			{ name: 'backgroundColor', className: 'background-color' },
+		],
+		{
+			contrastCheckers: [
+				{
+					backgroundColor: true,
+					textColor: true,
+					fontSize: fontSize.size,
+				},
+			],
+			colorDetector: { targetRef: ref },
+			colorPanelProps: {
+				initialOpen: true,
+			},
+		},
+		[ fontSize.size ]
+	);
+
+	const { align, showAvatar, showBio, byline } = props.attributes;
+
+	const avatarSizes = [];
+	if ( authorDetails ) {
+		forEach( authorDetails.avatar_urls, ( url, size ) => {
+			avatarSizes.push( {
+				value: size,
+				label: `${ size } x ${ size }`,
+			} );
+		} );
+	}
```

Comment:
Could do

```suggestion
	let avatarSizes = [];
	if ( authorDetails ) {
		avatarSizes = Object.keys( authorDetails.avatar_urls ).map( ( size ) => ( {
			value: size,
			label: `${ size } x ${ size }`,
		} ) );
	}
```

-- I personally prefer `.map` over `.forEach` since it's a bit more obvious to see that we're mapping one set of values to another.

Reply:
@ockham 's comment here was marked resolved, but I don't see there there were any revisions. I agree with the point to be using `map` here. Lodash's `map` can be slightly more expressive, at least since we're already using `forEach` from Lodash anyways:

```js
const avatarSizes = map( authorDetails?.avatar_urls, ( url, size ) => ( {
	value: size,
	label: `${ size } x ${ size }`,
} );
```